### PR TITLE
Allow expired billing teams to be deleted

### PR DESCRIPTION
**Pull request by hardillb**
part of #2559

## Description

Allow a team with expired billing e.g. at end of trial to be deleted even if it has instances/applications.

## Related Issue(s)

#2559 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Need to work out some tests for this

**Comment by hardillb** [comment]
Change to Billing.vue is just wrapping the whole thing in a `<div>` to clear VSCode lint warning

**Comment by hardillb** [comment]
The failing testcase is not something I've touched.

Wonder if it's a race condition?

File: forge/routes/api/team.js [hunk 1]
```diff
@@ -485,6 +485,40 @@ module.exports = async function (app) {
         // call to destroy the team will throw an error
         try {
             if (app.license.active() && app.billing) {
+                // It is possible the team has Instances but expired billing so
```
Comment (knolleary) [inline forge/routes/api/team.js:488]: We need to delete devices as well.

Comment (hardillb) [reply forge/routes/api/team.js:488]: Yep, will add that

File: forge/routes/api/team.js [hunk 2]
```diff
@@ -484,12 +484,54 @@ module.exports = async function (app) {
...
+                for (const instance of instances) {
+                    try {
+                        await app.containers.remove(instance)
+                    } catch (err) {
+                        if (err?.statusCode !== 404) {
+                            throw err
+                        }
+                    }
+
+                    // disconnect any devices from the Instance
```
Comment (knolleary) [inline forge/routes/api/team.js:500]: Given we're about to delete the devices, I don't think we should send them all a command that will cause them to call home and change their state.

**Review by knolleary — CHANGES_REQUESTED** [review]
One more tweak needed - no need to send devices a command if we're about to delete them anyway.

**Review by knolleary — APPROVED** [review]

**Pull request merged**